### PR TITLE
Update CODEOWNERS

### PR DESCRIPTION
**Pull request by jdunkerley**
Adding myself to GUI codeowners so can approve `CHANGELOG.md` changes by library team.

Happy to be removed once CHANGELOG in a new common location.

### Pull Request Description

No real changes just expanded code owners list for the `app/gui`.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/develop/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/develop/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

**Comment by mwu-tow** [comment]
Perhaps this won't be needed, as I'll be moving CHANGELOG.md to the repository root in the immediate future.

**Review by wdanilo — CHANGES_REQUESTED** [review]
@mwu-tow when it will happen?
@jdunkerley I think you should add yourself only to the CHANGELOG.md changes, not the whole GUI folder.

**Comment by mwu-tow** [comment]
> @mwu-tow when it will happen? @jdunkerley I think you should add yourself only to the CHANGELOG.md changes, not the whole GUI folder.

@wdanilo 
It is a part of https://github.com/enso-org/enso/pull/3241

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -13,7 +13,7 @@ Cargo.toml @MichaelMauderer @4e6 @mwu-tow @farmaazon
 # GUI
 /run @MichaelMauderer @wdanilo
 /build/ @MichaelMauderer @wdanilo
-/app/gui/ @MichaelMauderer @wdanilo @farmaazon @mwu-tow
+/app/gui/ @MichaelMauderer @wdanilo @farmaazon @mwu-tow @jdunkerley
```
Comment (farmaazon) [inline .github/CODEOWNERS]: As I'm a bit tired of being spammed with PRs which does not affect my area of knowledge, I'd consider removing the ownership from the CHANGELOG; usually someone competent should review PRs anyway.
```suggestion
/app/gui/ @MichaelMauderer @wdanilo @farmaazon @mwu-tow
/app/gui/CHANGELOG.md
```

**Comment by radeusgd** [comment]
I think we should also add me and James to `/test/` because this is the location where we keep tests for our libraries. Currently only @4e6 is the owner of that - but it seems there is no need to bother the engine team with every library changing PR (and obviously almost every PR will need to change tests).

**Comment by ekmett** [comment]
Meanwhile, can someone who _is_ approved finally go fix https://github.com/enso-org/enso/pull/3227 since it dares to touch that file so i can move on with my life?

**Review by wdanilo — APPROVED** [review]

**Review by PabloBuchu — APPROVED** [review]

**Pull request merged**